course/src/isometry.cc

```diff
@@ -0,0 +1,248 @@
+#include <isometry.h>
+
+namespace ekumen {
+  namespace math {
+
+  //Vector3
+
+  const Vector3 Vector3::kUnitX(1., 0., 0.);
+  const Vector3 Vector3::kUnitY(0., 1., 0.);
+  const Vector3 Vector3::kUnitZ(0., 0., 1.);
+  const Vector3 Vector3::kZero(0., 0., 0.);
+
+  Vector3 Vector3::operator += (const Vector3 & obj) const
+  {
+    return Vector3( x_ + obj.x(), y_ + obj.y(),z_ +  obj.z());
+  }
+
+  Vector3 Vector3::operator -= (const Vector3 & obj) const
+  {
+    return Vector3( x_ - obj.x(), y_ - obj.y(),z_ -  obj.z());
+  }
+
+  bool Vector3::operator == (const std::initializer_list<double> & obj) const
+  {
+    if (obj.size() != 3)
+    {
+      throw std::runtime_error("Size of list not valid");
+    }
+    if ( (obj.begin()[0] != x_) || (obj.begin()[1] != y_) || (obj.begin()[2] != z_) )
```

Comment:
Doing `obj.begin()[3]` is expensive. Implement this in some other way where you don't iterate more than once over each element of the initializer list.